packages/nc-gui/components/smartsheet/Kanban.vue

```diff
@@ -0,0 +1,493 @@
+    await updateKanbanMeta({
+      meta: stackMetaObj,
+    })
+  }
+}
+
+async function onMove(event: any, stackKey: string) {
+  if (event.added) {
+    const ele = event.added.element
+    ele.row[groupingField.value] = stackKey === 'uncategorized' ? null : stackKey
+    countByStack.value[stackKey] += 1
+    await updateOrSaveRow(ele)
+  } else if (event.removed) {
+    countByStack.value[stackKey] -= 1
+  }
+}
+
+const kanbanListScrollHandler = async (e: any) => {
+  if (e.target.scrollTop + e.target.clientHeight >= e.target.scrollHeight) {
+    const stackTitle = e.target.getAttribute('data-stack-title')
+    const pageSize = appInfo.defaultLimit || 25
+    const page = Math.ceil(formattedData.value[stackTitle].length / pageSize)
+    await loadMoreKanbanData(stackTitle, { offset: page * pageSize })
+  }
+}
+
+const kanbanListRef = (kanbanListElement: HTMLElement) => {
+  if (kanbanListElement) {
+    kanbanListElement.removeEventListener('scroll', kanbanListScrollHandler)
+    kanbanListElement.addEventListener('scroll', kanbanListScrollHandler)
+  }
+}
+
+const handleDeleteStackClick = (stackTitle: string, stackIdx: number) => {
+  deleteStackVModel.value = true
+  stackToBeDeleted.value = stackTitle
+  stackIdxToBeDeleted.value = stackIdx
+}
+
+const handleDeleteStackConfirmClick = async () => {
+  await deleteStack(stackToBeDeleted.value, stackIdxToBeDeleted.value)
+  deleteStackVModel.value = false
+}
+
+const handleCollapseStack = async (stackIdx: number) => {
+  groupingFieldColOptions.value[stackIdx].collapsed = !groupingFieldColOptions.value[stackIdx].collapsed
+  if (!isPublic.value) {
+    await updateKanbanStackMeta()
+  }
+}
+
+openNewRecordFormHook?.on(async (stackTitle) => {
+  const newRow = await addEmptyRow()
+  // preset the grouping field value
+  newRow.row = {
+    [groupingField.value]: stackTitle,
+  }
+  // increase total count by 1
+  countByStack.value.uncategorized += 1
+  // open the expanded form
+  expandForm(newRow)
+})
+
+onMounted(async () => {
+  // reset state to avoid from showing the previous stacks when switching kanban views
+  groupingFieldColOptions.value = []
+  formattedData.value = {}
+  // load kanban meta
+  await loadKanbanMeta()
+  // load kanban data
+  await loadKanbanData()
+  // update isMounted for below watcher
+  isMounted = true
+})
+
+watch(
+  () => shouldScrollToRight.value,
+  () => {
+    // use `isMounted` to avoid scrolling from switching from other views
+    // i.e. scroll only when a new option is added within kanban view
+    if (isMounted && shouldScrollToRight.value) {
+      // horizontally scroll to the end of the kanban container
+      kanbanContainerRef.value.scrollTo({
+        left: kanbanContainerRef.value.scrollWidth,
+        behavior: 'smooth',
+      })
+      // reset shouldScrollToRight
+      shouldScrollToRight.value = false
+    }
+  },
+)
+</script>
+
+<template>
+  <div class="flex h-full bg-white px-2">
```

Comment:
A couple of line breaks in the HTML template would help with readability.
Right now everything is just put next to each other so Hierarchies are pretty difficult to discern.